### ISSUE TITLE
ModelicaMatIO.c: Cast from pointer to integer of different size

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1880]
**Reported by otter on 14 Jan 2016 12:53 UTC**
DS reported that MinGW gcc reports the following warning for ModelicaMatIO.c:
```
../ModelicaMatIO.c: In function 'dopr':

../ModelicaMatIO.c:8445:51: warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]

                 fmtint (buffer, &currlen, maxlen, (long) strvalue, 16, min, max, flags);
```

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1880

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 14 Jan 2016 12:54 UTC**
When copying, a special character was removed from the Markdown interpreter, marking the error. Therefore, explicitely:
The "(long)strvalue" argument is claimed to cast from pointer to an integer of different size.

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 14 Jan 2016 13:02 UTC**
Hm, don't get this warning:
```
$ make TARGETDIR=win32
rm -f ModelicaStandardTables.o ModelicaMatIO.o ModelicaIO.o adler32.o compress.o crc32.o deflate.o gzclose.o gzlib.o gzread.o gzwrite.o infback.o inffast.o inflate.o inftrees.o trees.o uncompr.o zutil.o
rm -f *.a
rm -f ../../Library/win32/*.a
cc -DNDEBUG -DTABLE_SHARE=1 -DDUMMY_FUNCTION_USERTAB -O3 -I"../../C-Sources/zlib" -c -o ModelicaStandardTables.o ../../C-Sources/ModelicaStandardTables.c
ar -ru libModelicaStandardTables.a ModelicaStandardTables.o
d:\mingw-4.8.1\bin\ar.exe: creating libModelicaStandardTables.a
cp libModelicaStandardTables.a ../../Library/win32/libModelicaStandardTables.a
cc -DNDEBUG -O3 -I"../../C-Sources/zlib" -c -o ModelicaIO.o ../../C-Sources/ModelicaIO.c
ar -ru libModelicaIO.a ModelicaIO.o
d:\mingw-4.8.1\bin\ar.exe: creating libModelicaIO.a
cp libModelicaIO.a ../../Library/win32/libModelicaIO.a
cc -DNDEBUG -DHAVE_ZLIB=1 -O3 -I"../../C-Sources/zlib" -c -o ModelicaMatIO.o ../../C-Sources/ModelicaMatIO.c
ar -ru libModelicaMatIO.a ModelicaMatIO.o
d:\mingw-4.8.1\bin\ar.exe: creating libModelicaMatIO.a
cp libModelicaMatIO.a ../../Library/win32/libModelicaMatIO.a
cc -DNDEBUG -O3 -I"../../C-Sources/zlib" -c -o adler32.o ../../C-Sources/zlib/adler32.c
cc -DNDEBUG -O3 -I"../../C-Sources/zlib" -c -o compress.o ../../C-Sources/zlib/compress.c
cc -DNDEBUG -O3 -I"../../C-Sources/zlib" -c -o crc32.o ../../C-Sources/zlib/crc32.c
cc -DNDEBUG -O3 -I"../../C-Sources/zlib" -c -o deflate.o ../../C-Sources/zlib/deflate.c
cc -DNDEBUG -O3 -I"../../C-Sources/zlib" -c -o gzclose.o ../../C-Sources/zlib/gzclose.c
cc -DNDEBUG -O3 -I"../../C-Sources/zlib" -c -o gzlib.o ../../C-Sources/zlib/gzlib.c
cc -DNDEBUG -O3 -I"../../C-Sources/zlib" -c -o gzread.o ../../C-Sources/zlib/gzread.c
cc -DNDEBUG -O3 -I"../../C-Sources/zlib" -c -o gzwrite.o ../../C-Sources/zlib/gzwrite.c
cc -DNDEBUG -O3 -I"../../C-Sources/zlib" -c -o infback.o ../../C-Sources/zlib/infback.c
cc -DNDEBUG -O3 -I"../../C-Sources/zlib" -c -o inffast.o ../../C-Sources/zlib/inffast.c
cc -DNDEBUG -O3 -I"../../C-Sources/zlib" -c -o inflate.o ../../C-Sources/zlib/inflate.c
cc -DNDEBUG -O3 -I"../../C-Sources/zlib" -c -o inftrees.o ../../C-Sources/zlib/inftrees.c
cc -DNDEBUG -O3 -I"../../C-Sources/zlib" -c -o trees.o ../../C-Sources/zlib/trees.c
cc -DNDEBUG -O3 -I"../../C-Sources/zlib" -c -o uncompr.o ../../C-Sources/zlib/uncompr.c
cc -DNDEBUG -O3 -I"../../C-Sources/zlib" -c -o zutil.o ../../C-Sources/zlib/zutil.c
ar -ru libzlib.a adler32.o compress.o crc32.o deflate.o gzclose.o gzlib.o gzread.o gzwrite.o infback.o inffast.o inflate.o inftrees.o trees.o uncompr.o zutil.o
d:\mingw-4.8.1\bin\ar.exe: creating libzlib.a
cp libzlib.a ../../Library/win32/libzlib.a
```

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 14 Jan 2016 13:06 UTC**
MinGW should not compile function dopr at all as it has vsnprintf.

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 14 Jan 2016 13:12 UTC**
Ah, only forcing C99, i.e., `make TARGETDIR=win32 CFLAGS="-O3 -std=gnu99"` helps to explicitly get rid of dopr.

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 14 Jan 2016 13:23 UTC**
"%p" is not used in format srings of ModelicaMatIO, thus it is save to remove the lines 8443-8446 from ModelicaMatIO.

**Comment by modelica-trac-importer** [comment]
**Modified by beutlich on 14 Jan 2016 13:32 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 14 Jan 2016 22:05 UTC**
I'll fix it later since this basically is a non-issue.

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 15 Jan 2016 08:17 UTC**
@DS: Please test if this [ModelicaMatIO.c](https://github.com/tbeu/ModelicaStandardTables/blob/fix-snprintf/C-Sources/ModelicaMatIO.c) fixes the warning. The [Ubuntu build](https://drone.io/github.com/tbeu/ModelicaStandardTables/120) runs fine.

**Comment by modelica-trac-importer** [comment]
**Comment by peter&#x2e;nilsson on 15 Jan 2016 10:04 UTC**
Just a note about why you did not get the warning: It only occurs for 64-bit platform where sizeof(long) > sizeof(int).

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 15 Jan 2016 11:18 UTC**
I thought it is about sizeof(long) < sizeof(char\*).

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 15 Jan 2016 12:48 UTC**
@DS: Please reply if this MinGW64 warning is gone by the above mentioned file.

**Comment by modelica-trac-importer** [comment]
**Comment by peter&#x2e;nilsson on 15 Jan 2016 13:01 UTC**
The warnings is now gone, thanks.

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 15 Jan 2016 13:25 UTC**
Resolved by d24b63c0f7f404969ed92475aa30b06d32fda741.

**Comment by modelica-trac-importer** [comment]
**Modified by dietmarw on 23 Feb 2016 07:25 UTC**
